src/content/reference/fleet-management/migration_to_cluster_api.md

```diff
@@ -0,0 +1,44 @@
+  - What are the recommendations for a smooth migration?
+---
+
+In Giant Swarm we have used Kubernetes to build platforms since the beginning. On the first years there was no standard process provided by the community, and we develop our custom solution to manage the lifecycle of the clusters. It has worked well for us and our customers, but we learnt that many companies in the ecosystem were resolving the same problem over and over again, so we foster and push for joining forces and build a standard approach. [Cluster API]({{< relref "/reference/fleet-management/introduction" >}}) is backed by the Kubernetes community and covers different providers like AWS, Azure, GCP, and others.
+
+In this document we explain the migration path from our vintage system to the [Cluster API](https://cluster-api.sigs.k8s.io/) standard. The migration process was designed to ensure a smooth transition of customer workload clusters from our vintage generation to the new Cluster API (CAPI) flavour. This document provides an overview of the process, its requirements, and recommendations to ensure a successful migration.
+
+## Pre-migration requirements
+
+Before you begin the migration:
+
+1. Your cluster should be on a release version `>=20.0.0`.
+2. An AWS IAM role, named `giantswarm-{CAPI_MC_NAME}-capa-controller`, must be pre-created for the workload cluster's (WC) AWS account. [More information on how to do it in this guide](https://docs.giantswarm.io/vintage/getting-started/cloud-provider-accounts/cluster-api/aws/#overview).
+
+__Note:__ The `CAPI_MC_NAME` is the name of the management cluster (MC) where the Cluster API controllers are installed.
+
+## Recommendations for a smooth migration
+
+In preparation for migration, we recommend increasing the size of your "master" node instance type to two or three times its actual size (example, from `mx.large` to `mx.4xlarge`). This ensures the API server can efficiently handle the load during migration since there might be only one node to handle the traffic at certain points.
+
+## The migration process
+
+The migration process consists of several steps:
+
+1. __Initialization:__ Retrieval of necessary Kubernetes access credentials together with AWS credentials. At the same time, it creates a vault client to interact with the Vault instance that contains all security assets of the cluster.
+2. __Preparation:__ Migration of secrets to the CAPI management cluster, including CA certs, encryption provider secrets, and service account secrets. The migration scripts are created as a secret in the CAPI management cluster. Additionally, AWS credentials for the cluster are migrated by creating an `AWSClusterRoleIdentity` in the CAPI management cluster. There are certain operations performed to avoid conflicts during migration, such as disabling machine health check on the vintage cluster resources, scaling down the app operator for the migrated workload cluster, or cleaning up certain charts.
```

Comment:
```suggestion
2. __Preparation:__ Migration of secrets to the CAPI management cluster, including CA certs, encryption provider secrets, and service account secrets. Migration scripts are created as secrets in the CAPI management cluster. Additionally, AWS credentials for the cluster are migrated by creating an `AWSClusterRoleIdentity` in the CAPI management cluster. Certain operations are performed to avoid conflicts during migration, such as disabling machine health check on the vintage cluster resources, scaling down the app operator for the migrated workload cluster, or cleaning up certain charts.
```